components/service/glean/src/main/java/mozilla/components/service/glean/timing/TimingManager.kt

```diff
@@ -44,38 +54,44 @@ internal object TimingManager {
      * preserved.
      *
      * @param metricData The metric managing the timing. Used for error reporting.
-     * @param timerId The object to associate with this timing.
+     * @return a [GleanTimerId] representing the id to associate with this timing.
      */
-    public fun start(metricData: CommonMetricData, timerId: Any) {
+    fun start(metricData: CommonMetricData): GleanTimerId? {
         val startTime = getElapsedNanos()
 
+        var timerId: GleanTimerId?
+
         synchronized(this) {
+            timerId = timerIdCounter++
+
             val metricName = metricData.identifier
             uncommittedStartTimes[metricName]?.let { metricTimings ->
                 if (timerId in metricTimings) {
                     recordError(
```

Comment:
Shouldn't this also record an error if we have a series of calls "start, stop, start"?

Reply:
Mh, not at this layer I think. This is taken care of in the TimespanMetricType.